### ISSUE TITLE
[regression] Unable to remove dependencies between projects as specified in the docs via implicitDependencies

### ISSUE THREAD

**Issue by k1llabee** [nrwl/nx#15605]
### Current Behavior

In https://nx.dev/l/r/core-concepts/configuration, it specified that we can remove dependencies between projects (e.g. a lib seems to build an app while it does not have a connection via the project graph, so I'd like to explicitly remove it)

Finally, you can remove dependencies between projects. The following say even though the project imports 'mylib', it's not a dependency Nx should be concerned with.

{
  //... other project config
  "tags": [],
  "implicitDependencies": ["!mylib"]
}

Actually dependencies are still tracked.

### Expected Behavior

Feature works as documented.

### Steps to Reproduce

1.

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.13.2
   OS   : darwin x64
   npm  : 8.1.2
   
   nx                      : 15.8.5
   @nrwl/js                : 15.8.5
   @nrwl/jest              : 15.8.5
   @nrwl/linter            : 15.8.5
   @nrwl/workspace         : 15.8.5
   @nrwl/cli               : 15.8.5
   @nrwl/devkit            : 15.8.5
   @nrwl/eslint-plugin-nx  : 15.8.5
   @nrwl/nx-plugin         : 15.8.5
   @nrwl/tao               : 15.8.5
   typescript              : 4.9.5
   ---------------------------------------
   Community plugins:
   @jscutlery/semver : 2.30.0
   ngx-deploy-npm    : 5.0.0
   ---------------------------------------
   Local workspace plugins:
```

### Additional Information

https://github.com/nrwl/nx/issues/8322
same

**Comment by ShaunKT** [comment]
Hi, is there any update on this? A simple repo to highlight the bug: https://github.com/rdenner/nx-implicit-dependencies-broken

**Comment by AgentEnder** [comment]
Hey, there's a PR up to fix this (#15546)
